### PR TITLE
[stable/newrelic-infrastructure] Upgrade the New Relic k8s integration to v1.16.0

### PR DESCRIPTION
**Pull request by douglascamata**
#### Is this a new chart

No.

#### What this PR does / why we need it:

Upgrades the version of the New Relic Kubernetes integration.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Review by areina — APPROVED** [review]

File: stable/newrelic-infrastructure/templates/daemonset.yaml [hunk 1]
```diff
@@ -55,6 +55,10 @@ spec:
                   {{- end }}
             - name: "CLUSTER_NAME"
               value: {{ include "newrelic.cluster" . }}
+            {{- if .Values.disableKubeStateMetrics }}
+            - name: "DISABLE_KUBE_STATE_METRICS"
+              value: {{ .Values.kubeStateMetricsPodLabel | quote }}
```
Comment (sshishov) [inline stable/newrelic-infrastructure/templates/daemonset.yaml:60]: Here is a typo here. You are using incorrect value, it should be
```suggestion
              value: {{ .Values.disableKubeStateMetrics | quote }}
```

Comment (douglascamata) [reply stable/newrelic-infrastructure/templates/daemonset.yaml:60]: Thanks for noticing, @sshishov. I'll fix it right away.

Comment (douglascamata) [reply stable/newrelic-infrastructure/templates/daemonset.yaml:60]: Fix at #21712

**Pull request merged**